### PR TITLE
Issue 5725: Provide dlib_shared on desktop platforms for better tool support

### PR DESCRIPTION
**Pull request by JCash**

### CODE REVIEW

File: scripts/build.py [hunk 1]
```diff
@@ -511,6 +510,15 @@ def make_package_paths(root, platform, packages):
...
                 installed_packages.update(package_paths)
 
+        # For easier usage with the extender server, we want the linux protoc tool available
+        if target_platform in ('x86_64-darwin', 'x86_64-win32', 'x86_64-linux'):
+            protobuf_packages = filter(lambda x: "protobuf" in x, PACKAGES_HOST)
+            package_paths = make_package_paths(self.defold_root, 'x86_64-linux', protobuf_packages)
+            print("Installing %s packages " % 'x86_64-linux')
+            for path in package_paths:
+                self._extract_tgz(path, self.ext)
+            installed_packages.update(package_paths)
```
Comment (JCash) [inline scripts/build.py:520]: Since the extender server runs linux

File: scripts/build.py [hunk 2]
```diff
@@ -924,6 +932,12 @@ def archive_engine(self):
             gdc_bin = join(bin_dir, gdc_name)
             self.upload_to_archive(gdc_bin, '%s/%s' % (full_archive_path, gdc_name))
 
+        # upload libdlib_shared on desktop platforms
+        if self.target_platform in ['x86_64-linux', 'x86_64-darwin', 'x86_64-win32']:
+            lib_name = format_lib("dlib_shared", self.target_platform)[0]
+            lib_bin = join(bin_dir, lib_name)
+            self.upload_to_archive(lib_bin, '%s/%s' % (full_archive_path, lib_name))
```
Comment (JCash) [inline scripts/build.py:939]: In order to easily make a slim Defold SDK for testing locally, it's good if there is a built dlib_shared available from d.defold.com that you can easily download.

**Comment by britzl** [comment]
What's in dlib_shared by the way?

**Comment by JCash** [comment]
I think it's mainly used for the hash function, so that we can produce the same hashes that the engine expects.

**Pull request merged**